# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufSurface.cpp
# Commit: 89b662e27321
# Full Hash: 89b662e2732152591409f8c50551e075d86717a1
# Author: stransky <stransky@redhat.com>
# Date: 2025-01-09 04:16:57
# Description:
#   Bug 1940453 [Linux] Don't read empty DMABuf surfaces r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D233516
# ==============================================================================

diff -r e1021bf5dfd0 -r 89b662e27321 widget/gtk/DMABufSurface.cpp
--- a/widget/gtk/DMABufSurface.cpp	Wed Jan 08 19:07:20 2025 +0000
+++ b/widget/gtk/DMABufSurface.cpp	Wed Jan 08 19:12:43 2025 +0000
@@ -1732,6 +1732,11 @@
                                           gfx::SurfaceFormat aFormat) {
   LOGDMABUF(("DMABufSurfaceYUV::ReadIntoBuffer UID %d", mUID));
 
+  // We're empty, nothing to copy
+  if (!GetTextureCount()) {
+    return NS_ERROR_FAILURE;
+  }
+
   MOZ_ASSERT(aSize.width == GetWidth());
   MOZ_ASSERT(aSize.height == GetHeight());
 
